sim/druid/starfire.go

```diff
@@ -46,6 +46,10 @@ func (druid *Druid) newStarfireSpell() *core.Spell {
 			}
 		})
 	}
+	// If improved insect swarm and MF active, +3% crit chance
+	if druid.MoonfireDot.IsActive() && druid.Talents.ImprovedInsectSwarm > 0 {
+		effect.BonusSpellCritRating += 0.01 * float64(druid.Talents.ImprovedInsectSwarm)
```

Comment:
`BonusSpellCritRating` needs to be given in terms of crit rating, not crit chance. So you should change this to, `effect.BonusSpellCritRating += 1 * float64(druid.Talents.ImprovedInsectSwarm) * core.CritRatingPerCritChance`